src/utils/face++.js

```diff
@@ -0,0 +1,281 @@
+    const { faceset_token } = response
+    return faceset_token
+  } catch (e) {
+    console.error('create FaceSet error:', e)
+  }
+}
+
+const getDetailFaceSet = async () => {
+  let options = {
+    method: 'POST',
+    uri: getDetailFaceSetURL,
+    qs: {
+      api_key: faceKey,
+      api_secret: faceSecret,
+      faceset_token: faceSetToken
+    }
+  }
+  try {
+    limiter.removeTokens(1, async (err, remainingRequests) => {
+      let response = await request(options)
+      console.log('faceset details:', response)
+    })
+  } catch (e) {
+    console.error('get Detail FaceSet error:', e)
+  }
+}
+
+const removeAllFaces = async () => {
+  let options = {
+    method: 'POST',
+    uri: removeFaceURL,
+    qs: {
+      api_key: faceKey,
+      api_secret: faceSecret,
+      faceset_token: faceSetToken,
+      face_tokens: "RemoveAllFaceTokens"
+    }
+  }
+  try {
+    limiter.removeTokens(1, async (err, remainingRequests) => {
+      let response = await request(options)
+      console.log('removed all face tokens:', response)
+    })
+  } catch (e) {
+    console.error('get Detail FaceSet error:', e)
+  }
+}
+
+const addFace = async (faceToken) => {
+  let options = {
+    method: 'POST',
+    uri: addFaceURL,
+    qs: {
+      api_key: faceKey,
+      api_secret: faceSecret,
+      faceset_token: faceSetToken,
+      face_tokens: faceToken
+    },
+    json: true
+  }
+  try {
+    limiter.removeTokens(1, async (err, remainingRequests) => {
+      let response = await request(options)
+      console.log('addedFace res:', response)
+    })
+  } catch (e) {
+    console.error('detect FaceSet error:', e)
+  }
+}
+
+const detectFace = async (image_url) => {
+  let options = {
+    method: 'POST',
+    uri: detectFaceURL,
+    qs: {
+      api_key: faceKey,
+      api_secret: faceSecret,
+      image_url
+    },
+    json: true
+  }
+  try {
+    let response = await request(options)
+    const { faces, image_id } = response
+
+    //TODO: Need to store image_id in the database for when cron job runs. Identify which images are removed and which are newly added to decrease latency. Only update face_set with new images and delete face_set old images
+
+    //TODO: Batch face_tokens in arrays of 5 since there is only 1 QPS rate limit
+    console.log('face_token:', faces[0].face_token)
+    return faces[0].face_token
+  } catch (e) {
+    console.error('detect Face error:', e)
+  }
+}
+
+const setUserId = async (user_id, face_token) => {
+  console.log('setting UserID user_id:', user_id)
+  console.log('setting UserID face_token:', face_token)
+  let options = {
+    method: 'POST',
+    uri: setUserIdURL,
+    qs: {
+      api_key: faceKey,
+      api_secret: faceSecret,
+      face_token,
+      user_id
+    },
+    json: true
+  }
+  try {
+    let response = await request(options)
+    console.log('set userId response:', response)
+    return
+  } catch (e) {
+    console.error('detect Face error:', e)
+  }
+}
```

Comment:
Seem to be a lot of repeated logic in above functions. Consider making a helper function (closure) to generate these dynamically for different functionality.

Reply:
oh can you elaborate?

Reply:
The `try`/`catch` logic is largely the same, with the mainly the `options` differing. So maybe pull `options` into the parameters so you can just use one function that takes in different options. It'll probably be easier to modify if you need to change something instead of changing how all the functions work.